networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/rpc/core/methods/VersionedEth2RpcMethod.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2021 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.teku.networking.eth2.rpc.core.methods;
+
+import static com.google.common.base.Preconditions.checkArgument;
+
+import java.util.Comparator;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import tech.pegasys.teku.networking.eth2.rpc.core.Eth2IncomingRequestHandler;
+import tech.pegasys.teku.networking.eth2.rpc.core.Eth2OutgoingRequestHandler;
+import tech.pegasys.teku.networking.eth2.rpc.core.Eth2RpcResponseHandler;
+import tech.pegasys.teku.networking.eth2.rpc.core.encodings.RpcEncoding;
+import tech.pegasys.teku.spec.datastructures.networking.libp2p.rpc.RpcRequest;
+import tech.pegasys.teku.ssz.SszData;
+import tech.pegasys.teku.ssz.schema.SszSchema;
+
+public class VersionedEth2RpcMethod<
+        TRequest extends RpcRequest & SszData, TResponse extends SszData>
+    extends AbstractEth2RpcMethod<TRequest, TResponse> {
+  private final List<String> protocolIds;
+  private final Map<String, SingleProtocolEth2RpcMethod<TRequest, TResponse>> protocolToMethod;
+
+  public VersionedEth2RpcMethod(
```

Comment:
Any reason this should be public given we have the `create` method?

Reply:
This is just an artifact of having split up the larger PR - I cleaned this up at some later commit.  Will fix here as well 👍